deployments/helm/templates/deployment.yaml

```diff
@@ -57,11 +62,29 @@ spec:
             cpu: {{ .Values.deployment.resources.requests.cpu | quote}}
             memory: {{ .Values.deployment.resources.requests.memory }}
         {{- if not .Values.backup.enabled }}
+        {{- with .Values.deployment.env }}
+        env:
+          {{- toYaml . | nindent 12 }}
+        {{- end }}
         args:
           - --config_file 
           - /app/config/presets/{{ .Values.starknet.preset }}
           - --base_layer.node_url 
           - {{ .Values.base_layer_node_url }}
+          {{- if .Values.p2p.enabled }}
+          - --network.tcp_port
+          - {{ default "10000" .Values.p2p.config.networkTcpPort | quote }}
```

Comment:
not sure its a good idea to give default values inside the templates. if you specify the value in the default `values.yaml` you get the same result and its more explicit and readable.